### DISCUSSION TITLE
[ERROR] Could not resolve "vscode-textmate", "xterm*"..

### DISCUSSION THREAD

**Discussion by GSri30** [TypeFox/monaco-languageclient#506] [Q&A] [answered]
I was trying to integrate `"monaco-editor": "^0.39.0"` with `"monaco-languageclient": "6.2.0"`.
I was facing the below error
 
<img width="1190" alt="Screenshot 2023-06-29 at 3 04 04 PM" src="https://github.com/TypeFox/monaco-languageclient/assets/54752266/560312f2-e503-4df9-9e0a-f0a7830dadca">

I was getting the same error stack for the following
```
'vscode-textmate',
'vscode-oniguruma/release/onig.wasm',
'vscode-oniguruma',
'xterm',
'xterm-addon-canvas',
'xterm-addon-image',
'xterm-addon-search',
'xterm-addon-unicode11',
'xterm-addon-webgl',
'xterm-addon-serialize',
```

I'm using vite for bundle packaging. When I tried to add the above packages in `optimizeDeps: exclude {}`, I am getting the above same error at runtime.

Can anyone provide any leads?

```
package.json


...
"monaco-editor": "^0.39.0",
"monaco-languageclient": "6.2.0",
"vscode-languageclient": "~8.1.0",
"vscode-ws-jsonrpc": "3.0.0",
"vscode-json-languageservice": "~5.3.4",
"vscode-jsonrpc": "~8.1.0",
"vscode-uri": "~3.0.7"
```

**Comment by msujew** [comment, marked as answer]
Hey @GSri30, you seem to be using pnpm, which [doesn't install `peerDependencies` automatically](https://github.com/orgs/pnpm/discussions/3995).

However, the `monaco-vscode-api` [requires these packages](https://github.com/CodinGame/monaco-vscode-api/blob/d92ce1190a61f449af50539434f0f874e4fc81e9/package.json#L299-L311) as `peerDependencies`, which is exactly what you're seeing. Please install them into your `package.json` to resolve the error.

**Reply by CGNonofr** [reply]
Correct! they are peerDependencies mainly because they are not mandatory (you can chose to not use the textmate tokenizer)

Regarding xtern, it's supposed to only be required if you use the terminal service override. Is there an issue with it?
